From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: way-zer <himc.wicp@gmail.com>
Date: Sat, 2 Mar 2024 20:04:19 +0800
Subject: [PATCH] BUILD MindustryX
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

buildPlugins & Kotlin 1.9.20

patchArc改写成transform
way-zer <himc.wicp@gmail.com> on 2024/4/27 at 18:57
---
 build.gradle          | 11 ++---------
 core/build.gradle.kts |  1 +
 gradle.properties     |  2 +-
 settings.gradle       |  2 ++
 4 files changed, 6 insertions(+), 10 deletions(-)

diff --git a/build.gradle b/build.gradle
index 4599d8a8554059a9f7764d6b7f17daf85d1e0318..2b81d7a3fbe4d4fecffdf9ce97773fb558f71c23 100644
--- a/build.gradle
+++ b/build.gradle
@@ -13,8 +13,8 @@ buildscript{
 }
 
 plugins{
-    id "org.jetbrains.kotlin.jvm" version "1.6.0"
-    id "org.jetbrains.kotlin.kapt" version "1.6.0"
+    id "org.jetbrains.kotlin.jvm" version "1.9.20"
+    id "org.jetbrains.kotlin.kapt" version "1.9.20"
 }
 
 allprojects{
@@ -181,13 +181,6 @@ project(":core"){
         }
     }
 
-    gradle.taskGraph.whenReady{
-        //these are completely unnecessary
-        tasks.kaptGenerateStubsKotlin.onlyIf{ false }
-        tasks.compileKotlin.onlyIf{ false }
-        tasks.inspectClassesForKotlinIC.onlyIf{ false }
-    }
-
     //comp** classes are only used for code generation
     jar{
         exclude("mindustry/entities/comp/**")
diff --git a/core/build.gradle.kts b/core/build.gradle.kts
index 52d6972f5111e55502e3187e3c600e1843f2ec3f..6dfb14b6bb9401ec083a0f6994ed9bab8b303c97 100644
--- a/core/build.gradle.kts
+++ b/core/build.gradle.kts
@@ -4,6 +4,7 @@ import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 plugins {
     java
     kotlin("jvm")
+    id("mindustryX.buildExt")
 }
 sourceSets.main {
     java.setSrcDirs(listOf("src/", "../../src/", layout.buildDirectory.dir("/generated/sources/annotationProcessor/java/main")))
diff --git a/gradle.properties b/gradle.properties
index c2aab8787d6665397609b5b8bd930a3bb7b3b2b6..f5a1baf9d9c5b0644a6924f23ecd77aff664baee 100644
--- a/gradle.properties
+++ b/gradle.properties
@@ -18,7 +18,7 @@ kapt.use.worker.api=true
 # Compilation avoidance (see https://kotlinlang.org/docs/kapt.html#compile-avoidance-for-kapt-since-1-3-20)
 kapt.include.compile.classpath=false
 # I don't need to use the kotlin stdlib, so remove it to prevent extra bloat & method count issues
-kotlin.stdlib.default.dependency=false
+#kotlin.stdlib.default.dependency=false
 #needed for android compilation
 android.useAndroidX=true
 org.gradle.caching=true
diff --git a/settings.gradle b/settings.gradle
index 304e893b1657dd6d23f87a0c6c2939ac7c1d5c84..116bd48602f25dbaa4f6438934e42698abd4a0a4 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -9,6 +9,8 @@ if(JavaVersion.current().ordinal() < JavaVersion.VERSION_17.ordinal()){
     throw new Exception("!!! YOU MUST USE JAVA 17 OR ABOVE TO COMPILE AND RUN MINDUSTRY !!! Read the README. Your version: ${System.properties["java.version"]}")
 }
 
+includeBuild("../buildPlugins")
+
 include 'core', 'server', 'annotations', 'tools', 'tests'
 include 'desktop', 'ios'
 
